source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -1441,6 +1441,70 @@ void RtFileIo::writeDefinitions(const FilePath& documentPath, const RtTokenVec&
     writeDefinitions(ofs, names, options);
 }
 
+RtPrim RtFileIo::readPrim(std::istream& stream, const RtReadOptions* options)
+{
+    try
+    {
+        DocumentPtr document = createDocument();
+        XmlReadOptions xmlReadOptions;
+        if (options)
+        {
+            xmlReadOptions.applyFutureUpdates = options->applyFutureUpdates;
+        }
+        readFromXmlStream(document, stream, &xmlReadOptions);
+
+        PvtStage* stage = PvtStage::ptr(_stage);
+
+        RtReadOptions::ElementFilter filter = options ? options->elementFilter : nullptr;
+
+        // Keep track of renamed nodes:
+        ElementPtr elem = document->getChildren()[0];
+        PvtRenamingMapper mapper;
+        if (!filter || filter(elem))
+        {
+            if (elem->isA<NodeDef>())
+            {
+                return readNodeDef(elem->asA<NodeDef>(), stage)->prim();
```

Comment:
It would be safer to save into a PvtPrim variable first, and null check this, before accessing the prim() method to convert to an RtPrim. Same for each of the read functions below. Save to a local PvtPrim*, and check it's validity below before returning.